test/conformance/helpers/channel_tracing_test_helper.go

```diff
@@ -125,12 +126,15 @@ func setupChannelTracingWithReply(
 	// We expect the following spans:
 	// 1. Sending pod sends event to Channel (only if the sending pod generates a span).
 	// 2. Channel receives event from sending pod.
-	// 3. Channel sends event to transformer pod.
-	// 4. Transformer Pod receives event from Channel.
-	// 5. Channel sends reply from Transformer Pod to the reply Channel.
-	// 6. Reply Channel receives event from the original Channel's reply.
-	// 7. Reply Channel sends event to the logging Pod.
-	// 8. Logging pod receives event from Channel.
+	// 3. Channel Dispatcher span
+	// 4. Channel sends event to Mutator pod.
```

Comment:
Why 2 spans here? Are we sure this isn't an impl detail here and only 1 match should be done (the point 3)?

Reply:
I discovered the the span containing the cloudevent & tracing tags wasn't being tested previously. 

Removing the assertion for the span corresponding to `Channel sends event to Mutator pod` doesn't impact the issue this PR is attempting to address. However, It would just loose the test case to ensure the http status of the resultant request is not tested. I'd defer the importance of testing/not testing to folks with more context. 

Looking at [the broker tests](https://github.com/knative/eventing/blob/master/test/conformance/helpers/broker_tracing_test_helper.go#L135-L141), it doesn't appear that they are asserting for the presence of the span with the http status either.

So, I am happy to drop the assertions for spans corresponding to 4, 7 and 10. 

Does that make sense  @slinkydeveloper ?

Reply:
Sorry, I think i explained myself wrongly. Why do assert the span:

>  Channel Dispatcher span

Isn't that an implementation detail?

Reply:
Thats the span that contains the Open Telemetry Spec Attributes that this PR adds
Ditto for spans labeled 6 and 9